core/src/main/kotlin/org/evomaster/core/search/gene/sql/SqlLogSeqNumber.kt

```diff
@@ -0,0 +1,165 @@
+ *  for example, 16/B374D848.
+ *  Min value is 0/0
+ *  Max value is FFFFFFFF/FFFFFFFF
+ */
+class SqlLogSeqNumber(
+        /**
+         * The name of this gene
+         */
+        name: String,
+
+        /**
+         * The left part of 32 bits
+         */
+        private val leftPart: LongGene = LongGene("leftPart",
+                value = 0,
+                min = 0,
+                max = MAX_VALUE,
+                minInclusive = true,
+                maxInclusive = true),
+
+        /**
+         * The right part of 32 bits
+         */
+        private val rightPart: LongGene = LongGene("rightPart",
+                value = 0,
+                min = 0,
+                max = MAX_VALUE,
+                minInclusive = true,
+                maxInclusive = true),
+
+        ) : Gene(name, mutableListOf(leftPart, rightPart)) {
+
+    companion object {
+        val log: Logger = LoggerFactory.getLogger(SqlLogSeqNumber::class.java)
+
+        val MAX_VALUE = 2.toDouble().pow(32.toDouble()).toLong() - 1
+
+        private fun toHex(value: Long) = value.toString(16)
+
+    }
+
+
+    override fun getChildren(): MutableList<Gene> =
+            mutableListOf(leftPart, rightPart)
+
+    override fun copyContent(): Gene {
+        return SqlLogSeqNumber(
+                name = name,
+                leftPart = leftPart.copyContent() as LongGene,
+                rightPart = rightPart.copyContent() as LongGene
+        )
+    }
+
+    override fun copyValueFrom(other: Gene) {
+        if (other !is SqlLogSeqNumber) {
+            throw IllegalArgumentException("Invalid gene type ${other.javaClass}")
+        }
+
+        leftPart.copyValueFrom(other.leftPart)
+        rightPart.copyValueFrom(other.rightPart)
+    }
+
+    override fun containsSameValueAs(other: Gene): Boolean {
+        if (other !is SqlLogSeqNumber) {
+            throw IllegalArgumentException("Invalid gene type ${other.javaClass}")
+        }
+        return leftPart.containsSameValueAs(other.leftPart)
+                && rightPart.containsSameValueAs(other.rightPart)
+
+    }
+
+
+    override fun randomize(randomness: Randomness, forceNewValue: Boolean, allGenes: List<Gene>) {
+        log.trace("Randomizing ${this::class.java.simpleName}")
+        val genes: List<Gene> = listOf(leftPart, rightPart)
+        val index = randomness.nextInt(genes.size)
+        genes[index].randomize(randomness, forceNewValue, allGenes)
+    }
+
+    /**
+     * Forbid explicitly individual mutation
+     * of these genes
+     */
+    override fun candidatesInternalGenes(
+            randomness: Randomness,
+            apc: AdaptiveParameterControl,
+            allGenes: List<Gene>,
+            selectionStrategy: SubsetGeneSelectionStrategy,
+            enableAdaptiveGeneMutation: Boolean,
+            additionalGeneMutationInfo: AdditionalGeneMutationInfo?
+    ): List<Gene> {
+        return listOf()
```

Comment:
i guess this is fine for now. we will re-discuss it anyway in the refactoring meetings

Reply:
Agree.